openfisca_core/taxscales.py

```diff
@@ -195,7 +195,9 @@ def calc(self, base, factor = 1, round_base_decimals = None):
         base1 = np.tile(base, (len(self.thresholds), 1)).T
         if isinstance(factor, (float, int)):
             factor = np.ones(len(base)) * factor
-        thresholds1 = np.outer(factor, np.array(self.thresholds + [np.inf]))
+        # thresholds1 = np.outer(factor, np.array(self.thresholds + [np.inf]))
+        # changed to below to avoind NaN creation
+        thresholds1 = np.outer(factor + np.finfo(np.float).eps, np.array(self.thresholds + [np.inf]))
```

Comment:
That's high-level NumPy usage!

I'm afraid I don't understand the technique here, and if I had to touch this part of the code I'd be embarrassed.

You're here the NumPy expert :)